app/routes.py

```diff
@@ -1,2 +1,245 @@
-from flask import Blueprint
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, make_response, request
+from datetime import datetime
 
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@tasks_bp.route("", methods=["GET", "POST"])
+def handle_tasks():
+    if request.method == "GET":
+        sort_query = request.args.get("sort")
+        
+        if sort_query == "desc":
+            tasks = Task.query.order_by(Task.title.desc())
+        elif sort_query == "asc":
+            tasks = Task.query.order_by(Task.title.asc())            
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = []
+        for task in tasks:
+            tasks_response.append({
+                "id" : task.task_id,
+                "title" : task.title,
+                "description" : task.description,
+                "is_complete" : task.is_complete
+            })
+        return jsonify(tasks_response)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        title = request_body.get("title")
+        description = request_body.get("description")
+
+        if title is None or description is None:
+            return jsonify({"details" : "Invalid data"}), 400
+
+        try: 
+            completed_at = request_body["completed_at"]
+        except KeyError:
+            return jsonify({"details" : "Invalid data"}), 400
+
+        new_task = Task(title=title, description=description, completed_at=completed_at)
+
+        if new_task.completed_at:
+            new_task.is_complete = True
```

Comment:
If we remove `is_complete` and calculate its value when we make the result dictionary, we don't need the code to update it here.